lib/application.js

```diff
@@ -0,0 +1,186 @@
+var url = require("url");
+var qs = require("qs");
+
+/**
+ * Application
+ * @constructor
+ * @param {Object} client Catapult client
+ */
+
+var Application = function (client) {
+	/**
+	 * List the user's applications
+	 * @param {Object} params Parameters for filtering applications
+	 * @param {Number} [params.size] The size of each page requested (Max size: 1000)
+	 * @param {Function} [callback] A callback for the list of applications
+	 * @return {MessageResponse} A promise for the list of applications
+	 */
+	this.list = function (params, callback) {
+		var self = this;
+		return client.makeRequest({
+			path   : "applications",
+			method : "GET",
+			qs     : params
+		})
+		.then(function (response) {
+			var applicationListResponse = {
+				applications : response.body
+			};
+			var links = response.headers.link.split(",");
+			var nextLink = links[1];
```

Comment:
If sizeof links is only 1, then `nextLink = links[1]` will be undefined and be caught by the following line.